### ISSUE TITLE
Suggestions for search criteria for elliptic curves

### ISSUE THREAD

**Issue by jwj61** [LMFDB/lmfdb#6176]
From the feedback page:

I have been looking for elliptic curves with torsion point (0,0) and I had to do it manually which makes things much difficult. It is only as a suggestion but it would be extremely if the filtering could be done to a greater extent, for example for the torsion subgroup to contain a particular set of points. In edition to that, filtering out by the regulator and also the minimal discriminant can be very helpful when thinking about the Lang's Conjecture. Thank you so much though otherwise it is a great website!

**Comment by roed314** [comment]
* It is already possible to search on regulator and (minimal) discriminant, though perhaps it's not clear that "Discriminant" means minimal discriminant.
* There's no good way to filter on containing a specific torsion point, for three reasons.  First, torsion generators are stored in the `ec_mwbsd` table, meaning that they're not available in `ec_curvedata` where we're searching.  Second, we only store generators rather than the full set of points, so when the torsion subgroup is something other than Z/2 the particular point desired may not even show up stored in the data even though it is a torsion point.  Finally, searching for a list containing a triple may be slow in postgres (I'm not sure a GIN index applies here).

I'm also confused about the mathematical meaningfulness of a specific point lying on the curve, when we're storing isomorphism classes of curves.  Is it just (0,0) that's important, or is it relevant to be able to filter on any particular point?

In summary, I think we should close this issue, though I'm happy to get more context from the original requestor.

**Comment by AndrewVSutherland** [comment]
I agree.  And for the benefit of whoever posted this request, I think any elliptic curve with a torsion point $$P$$ of order $$N>1$$ has a model with $$P=(0,0)$$.  For $$N=2$$ this is clear (put $$E$$ in short Weierstrass form and translate the $$x$$-coordinate to zero), for $$N=3$$ use Kubert's parameterization $$y^2 + a_1xy+ a_3y = x^3$$ with $$a_1^3a_3^3 - 27a_3^4 \ne 0$$, and for $$N>3$$ you can put the curve in Tate normal form $$y^2+(1-c)xy-by=x^3-bx^2$$ for some $$b,c$$ (see [Kubert's paper](https://londmathsoc.onlinelibrary.wiley.com/doi/abs/10.1112/plms/s3-33.2.193) for the constraints on $$b,c$$ for $$N=4,5,6,7,8,9,10,12$$).  In each case you can compute a map to the $$j$$-line in terms of $$b,c$$ and for any particular $$E$$ with a torsion point $$P$$ of order $$N$$ you can find $$(b,c)$$ mapping to $$j(E)$$ and (modulo twisting) get a model for $$E$$ that puts $$P$$ at $$(0,0)$$.
